caffe2/python/operator_test/mean_op_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2016-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##############################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from caffe2.python import core
+from hypothesis import given
+import hypothesis.strategies as st
+import caffe2.python.hypothesis_test_util as hu
+import numpy as np
+
+import unittest
+
+
+class TestMean(hu.HypothesisTestCase):
+    @given(
+        k=st.integers(1, 5),
+        n=st.integers(1, 10),
+        m=st.integers(1, 10),
+        in_place=st.booleans(),
+        engine=st.sampled_from(["", "CUDNN"]),
+        **hu.gcs
+    )
+    def test_mean(self, k, n, m, in_place, engine, gc, dc):
+        input_names = []
+        input_vars = []
+
+        for i in range(k):
+            X_name = 'X' + str(i)
+            input_names.append(X_name)
+            var = np.random.randn(n, m).astype(np.float32)
```

Comment:
Since you do np.random.randn here, it is probably better to have a seed generated by hypothesis, and set np random seed before running the generation.